Build failure for Infineon CYT4BFC 320 BGA pin package

### Describe the bug

Commit https://github.com/zephyrproject-rtos/zephyr/pull/97771/commits/56697d62725ad90df226e1326dc23eaf28cd7eb8 has introduced compilation failure while building for Infineon CYT4BFC supporting 320 BGA pin package.
This issue is uncovered, while adding support for CYTVII-B-H-8M-CPU board for CYT4BFC which includes cyhal_xmc7200_320_bga.c from hal-infineon pin package. 
Issue is in cmakelist.txt, which includes cyhal_xmc7200_320_bga.c.c instead of cyhal_xmc7200_320_bga.c

<img width="1197" height="422" alt="Image" src="https://github.com/user-attachments/assets/d726a347-b9d0-4eb8-b85c-60fe6760f978" />

### Regression

- [x] This is a regression.

### Steps to reproduce

_No response_

### Relevant log output

```shell
CMake Error at C:/Users/khanapur/linumiz/zephyr/cmake/modules/extensions.cmake:536 (target_sources):
  Cannot find source file:

    C:/Users/khanapur/linumiz/modules/hal/infineon/mtb-hal-cat1/COMPONENT_CAT1C/source/pin_packages/cyhal_xmc7200_320_bga.c.c
Call Stack (most recent call first):
  C:/Users/khanapur/linumiz/zephyr/cmake/modules/extensions.cmake:2068 (zephyr_library_sources)
  C:/Users/khanapur/linumiz/zephyr/modules/hal_infineon/mtb-hal-cat1/CMakeLists.txt:70 (zephyr_library_sources_ifdef)
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

--- a/modules/hal_infineon/mtb-hal-cat1/CMakeLists.txt
+++ b/modules/hal_infineon/mtb-hal-cat1/CMakeLists.txt
@@ -74,7 +74,7 @@ zephyr_library_sources_ifdef(CONFIG_SOC_PACKAGE_XMC7200_176_TEQFP
 zephyr_library_sources_ifdef(CONFIG_SOC_PACKAGE_XMC7200_272_BGA
   ${hal_cat1c_dir}/source/pin_packages/cyhal_xmc7200_272_bga.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_PACKAGE_XMC7200_320_BGA
-  ${hal_cat1c_dir}/source/pin_packages/cyhal_xmc7200_320_bga.c.c)
+  ${hal_cat1c_dir}/source/pin_packages/cyhal_xmc7200_320_bga.c)
 
 zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_CYW20829
   ${hal_cat1b_dir}/source/triggers/cyhal_triggers_cyw20829.c)

